File: Easy/20_Valid_Parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """

        lst = []

        parentheses = {')': '(', '}': '{', ']': '['}
        for char in s:
            if char in parentheses:
                if len(lst)>0 and lst[-1] == parentheses[char]:
                    lst.pop()
                else:
                    return False
            else:
                lst.append(char)


        if len(lst) == 0:
            return True
        else:
            False



if __name__ == "__main__":
    assert Solution().isValid('([])') == True

File: Easy/20_Valid_Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let brackets = {
        '(': ')',
        '{': '}',
        '[': ']'
    }
    let stack = [];
    
    for(let i=0; i<s.length; i++) {
        if (brackets[s[i]]) {
            stack.push(brackets[s[i]]);
        }
        else {
            if (stack.pop() !== s[i]) return false;
        }
    }
    return !stack.length;
};
